### PR TITLE
Adding zipline dependencies (trading-calendars, python-interface, iso4217)

### PR DESCRIPTION
**Pull request by richafrank**
Includes zipline, trading-calendars, python-interface, and iso4217

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by gerrymanoim** [comment]
I'm okay being a maintainer.

**Comment by llllllllll** [comment]
I am okay being a maintainer.

**Comment by jmccorriston** [comment]
I am okay being a maintainer.

**Comment by jbredeche** [comment]
Ok fine I'm willing to be a maintainer

File: recipes/zipline/LICENSE
Comment (synapticarbors) [inline recipes/zipline/LICENSE:1]: This license file is already included in the source distribution, so you can remove it here.

File: recipes/trading-calendars/LICENSE
Comment (synapticarbors) [inline recipes/trading-calendars/LICENSE:1]: This license file is already included in the source distribution, so please remove it here

File: recipes/python-interface/meta.yaml
Comment (synapticarbors) [inline recipes/python-interface/meta.yaml:41]: Since you are manually packaging the license file, please add the following comment as a reminder to check whether the license file is included in future releases in the source distribution:

```yaml
...
license_family: APACHE
# License file manually packaged. See https://github.com/ssanderson/python-interface/pull/40
license_file: LICENSE
```

I opened a PR at https://github.com/ssanderson/python-interface/pull/40 to try to deal with this

**Review by synapticarbors — CHANGES_REQUESTED** [review]
Even thought `staged-recipes` correctly handles multiple inter-dependent recipes in a single PR, once this is merged, you are likely to see build issues when not all of the dependent packages have hit the conda-forge CDN yet:

https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=multiple#adding-multiple-packages-at-once

It might be worth it to split the `zipline` recipe off into its own PR to avoid this.

**Comment by richafrank** [comment]
Thanks @synapticarbors . 

I'll make those license changes and split out zipline. Will open another PR for that...

**Comment by jdricklefs** [comment]
I also approve being listed as a maintainer.

**Comment by richafrank** [comment]
@rsheftel Are you ok being added here as a maintainer of trading-calendars?

**Comment by rsheftel** [comment]
Yes

**Comment by richafrank** [comment]
@synapticarbors Updated as you suggested (I believe). The zipline-only PR is https://github.com/conda-forge/staged-recipes/pull/12569, and we have ok's from the listed maintainers. Thanks so much.

**Comment by synapticarbors** [comment]
@richafrank -- One quick thing that I apologize for not spotting previously. For `iso4217` and `python-interface`, conda-forge does not build any packages for `py < 3.6` unless you specifically need python 2.7 support (See: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-add-python-2-7). Therefore, I think we could drop all of the `py2k` and `py<34` selectors, and make those two packages `noarch: python` and then just changes `python` to `python >=3.6` under `host` and `run`.

**Comment by richafrank** [comment]
Oh, that's a great call. So much nicer to have the noarch builds. I just updated them as you suggested.

**Review by synapticarbors — APPROVED** [review]

**Pull request merged**
